### PR TITLE
Parameterized user home dir. With this change any user, not only xf11i…

### PR DESCRIPTION
**Pull request by oksanagit**
…d can ran scans. Tested as oksana

### CODE REVIEW

File: startup/99-bluesky.py [hunk 1]
```diff
@@ -300,7 +301,8 @@ def event(self, doc):
         return doc
 
 
-directory = '/home/xf11id/specfiles/'
+## directory = '/home/xf11id/specfiles/'
+directory = os.path.expanduser("~") + "/specfiles/"
```
Comment (mrakitin) [inline startup/99-bluesky.py]:
```suggestion
directory = os.path.join(os.path.expanduser("~"), "specfiles/")
```

**Review by mrakitin — COMMENTED** [review]
Thanks for the contribution, @oksanagit! I left a minor improvement suggestion to stich the file path in the OS-independent way.

I noticed that the same hardcoded variable is used below within a function (see https://github.com/NSLS-II-CHX/profile_collection/pull/15/files#diff-30351e59d8b68c700d5d4c3fc2666a1dR322). I think we can just reuse the `directory` there. Could you please also make this change? Thanks.

**Comment by oksanagit** [comment]
I made a change to line 329    from  
full_path='/home/xf11id/specfiles/'+name+'.spec'
to 
    full_path=os.path.join(os.path.expanduser("~", "specfiles/", name, '.spec')
How I can test it? The way i start bsui, this line never gets executed.

File: startup/99-bluesky.py [hunk 2]
```diff
@@ -324,7 +326,7 @@ def new_spec_file(name):
     - name= xyz .spec will be added automatically
     calling sequence: new_spec_file(name='xyz')
     """
-    full_path='/home/xf11id/specfiles/'+name+'.spec'
+    full_path=os.path.join(os.path.expanduser("~", "specfiles/", name, '.spec')
```
Comment (mrakitin) [inline startup/99-bluesky.py]:
```suggestion
    full_path = os.path.join(directory, f'{name}.spec')
```

**Review by mrakitin — CHANGES_REQUESTED** [review]
In the original implementation there was a syntax error on line 329 with the missing closing parenthesis. I suggest to use the defined earlier `directory` var. The extension should be joined separately, as `os.path.join(...)` will join it with the path separators, `/` for Linux.

**Review by mrakitin — APPROVED** [review]

**Comment by mrakitin** [comment]
I think it looks correct now.

@afluerasu, just to let you know that we fixed one hard-coded path here, to make it more portable.

**Comment by danielballan** [comment]
Thanks @oksanagit!

**Pull request merged**